lib/pbench/test/functional/server/test_put.py

```diff
@@ -117,6 +119,36 @@ def test_bad_name(server_client: PbenchServerClient, login_user):
             == "File extension not supported, must be .tar.xz"
         )
 
+    @staticmethod
+    def test_archive_only(server_client: PbenchServerClient, login_user):
+        """Try to upload a new dataset with the archiveonly option set, and
+        validate that it doesn't get enabled for unpacking or indexing."""
+        tarball = next(iter((TARBALL_DIR / "bad").glob("*.tar.xz")))
+        md5 = Dataset.md5(tarball)
+        response = server_client.upload(tarball, metadata={"server.archiveonly:y"})
+        assert (
+            response.status_code == HTTPStatus.CREATED
+        ), f"upload returned unexpected status {response.status_code}, {response.text}"
+        metadata = server_client.get_metadata(
+            md5, ["dataset.operations", "server.archiveonly"]
+        )
+        assert metadata["server.archiveonly"] is True
+
+        # NOTE: we could wait here; however, the UNPACK operation is only
+        # enabled by upload, and INDEX is only enabled by UNPACK: so if they're
+        # not here immediately after upload, they'll never be here.
+        operations = metadata["dataset.operations"]
+        assert "UNPACK" not in operations
+        assert "INDEX" not in operations
+        assert operations["UPLOAD"]["state"] == "OK"
+        assert operations["BACKUP"]["state"] in ("OK", "READY", "WORKING")
+
+        # Delete it so we can run the test case again without manual cleanup
```

Comment:
`test_delete_all()` _would_ have removed this dataset, right?  So, the real reason for removing it here is that it will never get indexed, which will cause `test_index_all()` to fail, unless we make `check_indexed` smart enough to skip reporting it as "not indexed".

(This is a subtle suggestion to remove the deletion from here, and to enhance `check_indexed()`...failing that, it's a suggestion to point out the additional importance of removing the data set here....)

Reply:
No, `test_delete_all` very carefully works from the tarball directory listing, and this is outside it. (Because otherwise it would have been a duplicate.) The logic changes in `test_index_all` would have been messy since we're *not* looking for indexing; and I didn't see any point since only the backup state will change, and we cover that elsewhere.